Swing Kit Docs (#1730)
vaadin/docs: articles/tools/swingkit/usage/swingbase.asciidoc

---
title: Swing Usage for Swing Kit
order: 4

> layout: page

Reviewer: ```suggestion
description: How to add Vaadin views to your Swing applications and how to communicate between them.
```

---

= Swing Usage for Swing Kit

== JVaadinPanel

> A Vaadin view can be rendered as a Swing Component like so:

Reviewer: ```suggestion
A Vaadin view can be rendered as a Swing component like so:
```

[source,java]
----
JVaadinPanel vaadinPanel = SwingVaadinClient.getBuilder().build(vaadinViewUrl);
----

> This `JVaadinPanel` extends the `Panel`. It can be added using standard swing layouts.

Reviewer: ```suggestion
The [classname]`JVaadinPanel` class extends the [classname]`Panel` component, and can be added to standard Swing layouts.
```

> You can check when the page has been fully loaded by using the `isReady()` method.

Reviewer: ```suggestion
You can check when the page has been fully loaded by using the [methodname]`isReady()` method.
```

== Handle Flow Swing Kit Events

> Events emitted from the server can be handled with `addEventListener`, as you see here:

Reviewer: ```suggestion
Events emitted from the Flow application can be handled with [methodname]`addEventListener()`.
For example:
```
